pkg/local/diff.go

```diff
@@ -385,19 +391,24 @@ func ListRemote(ctx context.Context, client apigen.ClientWithResponsesInterface,
 	return nil
 }
 
-func includeLocalFileInDiff(info fs.FileInfo, includeDirs bool) bool {
+var ignoreFileList = []string{
+	IndexFileName,
+	".DS_Store",
+}
+
+func includeLocalFileInDiff(info fs.FileInfo, cfg Config) (bool, error) {
 	if info.IsDir() {
-		return includeDirs
-	} else {
-		switch info.Name() {
-		case IndexFileName, ".DS_Store":
-			return false
-		default:
-			return true
+		return cfg.IncludePerm, nil
+	}
+	if !info.Mode().IsRegular() {
+		if !cfg.SkipIrregularFiles {
+			return false, fmt.Errorf("%s: %w", info.Name(), fileutil.ErrNotARegularFile)
 		}
+		return false, nil
```

Comment:
Can in Irregular file be in the `ignoreFileList`?

Reply:
The ignore file list is a list of files we decided to ignore. It should not contain any irregular files and if at some point it would - we'd need to resolve that internally (it's not something controlled by the user)